src/platform/zephyr/lib/dai.c

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Jaska Uimonen <jaska.uimonen@linux.intel.com>
+
+#include <device.h>
+#include <drivers/dai.h>
+#include <sof/audio/component.h>
+#include <sof/drivers/timestamp.h>
+#include <sof/lib/dai.h>
+#include <sof/lib/dma.h>
+#include <ipc/dai.h>
```

Comment:
do we really want platform/zephyr? If Zephyr is becoming the default RTOS for SOF shouldn't this just be src/lib/zephyr_dai.c?

Reply:
That's a very good discussion. I'm good for this as well. It kind of makes sense that we have a lot of platform specific code in src/platform that calls "dai_init()" to register all the dais for a given platform. When you build for Zephyr, this same is done with a different logic, and it kind of makes sense to have this in src/platform.